File: packages/core/src/marginalia_core/application/frontend/capabilities.py
"""Capability DTOs exposed to frontends."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class BackendCapabilities:
    """Backend-declared features and supported contract surface."""

    protocol_version: int
    commands: tuple[str, ...]
    queries: tuple[str, ...]
    transports: tuple[str, ...]
    frontend_event_stream_supported: bool
    dictation_enabled: bool
    rewrite_enabled: bool
    summary_enabled: bool

File: crates/marginalia-core/src/frontend/capabilities.rs
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct BackendCapabilities {
    pub protocol_version: u32,
    pub commands: Vec<String>,
    pub queries: Vec<String>,
    pub transports: Vec<String>,
    pub frontend_event_stream_supported: bool,
    pub dictation_enabled: bool,
    pub rewrite_enabled: bool,
    pub summary_enabled: bool,
}
